src/server/database/Database/Field.cpp

```diff
@@ -28,282 +30,310 @@ Field::Field()
     meta = nullptr;
 }
 
-Field::~Field() = default;
-
-uint8 Field::GetUInt8() const
+namespace
 {
-    if (!data.value)
-        return 0;
-
-#ifdef ACORE_STRICT_DATABASE_TYPE_CHECKS
-    if (!IsType(DatabaseFieldTypes::Int8))
+    template<typename T>
+    constexpr T GetDefaultValue()
     {
-        LogWrongType(__FUNCTION__);
-        return 0;
+        if constexpr (std::is_same_v<T, bool>)
+            return false;
+        else if constexpr (std::is_integral_v<T>)
+            return 0;
+        else if constexpr (std::is_floating_point_v<T>)
+            return 1.0f;
+        else if constexpr (std::is_same_v<T, std::vector<uint8>> || std::is_same_v<std::string_view, T>)
+            return {};
+        else
+            return "";
     }
-#endif
-
-    if (data.raw)
-        return *reinterpret_cast<uint8 const*>(data.value);
-    return static_cast<uint8>(strtoul(data.value, nullptr, 10));
-}
-
-int8 Field::GetInt8() const
-{
-    if (!data.value)
-        return 0;
 
-#ifdef ACORE_STRICT_DATABASE_TYPE_CHECKS
-    if (!IsType(DatabaseFieldTypes::Int8))
+    template<typename T>
+    inline bool IsCorrectFieldType(DatabaseFieldTypes type)
     {
-        LogWrongType(__FUNCTION__);
-        return 0;
+        // Int8
+        if constexpr (std::is_same_v<T, bool> || std::is_same_v<T, int8> || std::is_same_v<T, uint8>)
+        {
+            if (type == DatabaseFieldTypes::Int8)
```

Comment:
Question, why not just do

```
return type == DatabaseFieldTypes::Int8;
```

Else the function will just run and validate the next 10 IF statements for no reason

Reply:
Need check db type and c++ type